pkg/liquidity-mining/test/ChildChainGaugeCheckpointer.test.ts

```diff
@@ -0,0 +1,65 @@
+import { ethers } from 'hardhat';
+import { Contract } from 'ethers';
+
+import { sharedBeforeEach } from '@balancer-labs/v2-common/sharedBeforeEach';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+
+describe('ChildChainGaugeCheckpointer', () => {
+  let gaugeFactory: Contract;
+  let gauge1: Contract;
+  let gauge2: Contract;
+  let checkpointer: Contract;
+  let admin: SignerWithAddress, user1: SignerWithAddress, other: SignerWithAddress;
+  let registry: Contract;
+
+  before('setup signers', async () => {
+    [, admin, user1, other] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy gauge ChildChainGaugeCheckpointer', async () => {
```

Comment:
(not strong opinion) maybe consider splitting this block in two smaller ones: one that just deploys the base contracts up to the checkpointer, and one that sets up the permissions, adds the factory to the minter, deploys the gauges and adds them to the registry.

Reply:
Splitting one `sharedBeforeEach` into multiple would require declaring many function vars at the test scope level which would pollute the list of vars used in the test. Also the whole code being inside one function makes it more obvious that it is executed sequentially.